### PR TITLE
feat: Added logging

### LINKED ISSUE #73: `SetTag` function uses print function instead of logger
### Specifications

* Package Version: v0.6.0
* InfluxDB Version: V3
* Platform: 

### Code sample to reproduce problem

```go
WriteData(context.Background(), []any{measurement})
```

Some measurement has empty tags and some don't. The Write method should use a logger to warn for the empty values but instead it prints the warning which we cannot control over. This will result in lot of logs.

### Expected behavior

Should only print the empty tag as warning

### Actual behavior

```
2024-04-02T18:53:19.013+02:00	Empty tags has no effect, tag [*****], measurement [*****]
2024-04-02T18:53:19.013+02:00	Empty tags has no effect, tag [*****], measurement [*****]
2024-04-02T18:53:19.013+02:00	Empty tags has no effect, tag [*****], measurement [*****]
2024-04-02T18:53:19.013+02:00	Empty tags has no effect, tag [*****], measurement [*****]
2024-04-02T18:53:19.013+02:00	Empty tags has no effect, tag [*****], measurement [*****]
```

### Additional info
Code block https://github.com/InfluxCommunity/influxdb3-go/blob/main/influxdb3/point_values.go#L73
_No response_

### PR DESCRIPTION
**Pull request by thulasirajkomminar**
Closes #73

## Proposed Changes

_Briefly describe your proposed changes:_

Added log package to solve issue https://github.com/InfluxCommunity/influxdb3-go/issues/73

## Checklist

- [x] CHANGELOG.md updated
- [x] Rebased/mergeable
- [x] A test has been added if appropriate
- [x] Tests pass
- [x] Commit messages are [conventional](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Sign [CLA](https://www.influxdata.com/legal/cla/) (if not already signed)

### CODE REVIEW

**Review by bednar — CHANGES_REQUESTED** [review]
@thulasirajkomminar thanks for your PR 👍 

We want to keep the client source base as simple as possible, so the logging infrastructure should be also simple. 

What do you think about add simple logger implementation like following code? 

`client.go`:

```go
var logger *log.Logger

var logLevels = map[string]bool{
	"debug": true,
	"info":  true,
	"warn":  true,
	"error": true,
}

// ConfigureLogger initializes logger with specified log levels.
// Supported levels: debug, info, warn, error.
func ConfigureLogger(levels map[string]bool) {
	for level, enabled := range levels {
		if _, exists := logLevels[level]; exists {
			logLevels[level] = enabled
		}
	}
}

```

initialialize this logger in `New` function:

```go
logger = log.New(os.Stdout, "influxdb3-go: ", log.LstdFlags)
ConfigureLogger(map[string]bool{"debug": true, "info": true, "warn": false, "error": true})
```

and use it in `point_values.go`:

```go
func (pv *PointValues) SetTag(name string, value string) *PointValues {
	if value == "" {
		if enabled, ok := logLevels["debug"]; ok && enabled {
			logger.Printf("Empty tags has no effect, tag [%s], measurement [%s]\n", name, pv.MeasurementName)
		}
	} else {
		pv.Tags[name] = value
	}
	return pv
}
```

also you can enhance the above implementation with enum-like-constants:

```go
type LogLevel int

const (
	Debug LogLevel = iota
	Info
	Warn
	Error
)
```

**Comment by thulasirajkomminar** [comment]
@bednar Thought about just adding a simple logger but then I wanted to follow the V2 standards :) . So I like this simple implementation so modified my commit.

**Comment by bednar** [comment]
> Should we add log level to `ClientConfig` or should we override the defaults like this on the user side `influxdb3.ConfigureLogger(map[string]bool{"debug": false, "info": true, "warn": false, "error": true})`

Good catch! 👍 You are right it should be a part of ClientConfig.

**Comment by thulasirajkomminar** [comment]
> > Should we add log level to `ClientConfig` or should we override the defaults like this on the user side `influxdb3.ConfigureLogger(map[string]bool{"debug": false, "info": true, "warn": false, "error": true})`
> 
> Good catch! 👍 You are right it should be a part of ClientConfig.

@bednar could you verify?

File: influxdb3/point_values.go
Comment (bednar) [inline influxdb3/point_values.go:73]: What about keep it super simple and use this:
```suggestion
		slog.Debug(fmt.Sprintf("Empty tags has no effect, tag [%s], measurement [%s]", name, pv.MeasurementName))
```

The user can configure logger by:

```golang
// enable debug logging
opts := &slog.HandlerOptions{
        Level: slog.LevelDebug,
}

logger := slog.New(slog.NewJSONHandler(os.Stdout, opts))

slog.SetDefault(logger)
```

**Review by bednar — CHANGES_REQUESTED** [review]
For simplify our codebase we can use default logger:

**Comment by thulasirajkomminar** [comment]
> For simplify our codebase we can use default logger:

@bednar Could you have a look?

File: CHANGELOG.md
Comment (bednar) [inline CHANGELOG.md:6]:
```suggestion
2. [#74](https://github.com/InfluxCommunity/influxdb3-go/pull/74): Use `log/slog` to print debug information instead of `fmt.Printf`
```

**Review by bednar — CHANGES_REQUESTED** [review]
@thulasirajkomminar thanks again for your PR 👍 

There is only little change in CHANGELOG.md. After that we will be able to merge PR:

**Review by bednar — APPROVED** [review]
LGTM 👍 

We will merge the PR after we fix our CI to support PRs from forked repository.

**Comment by alespour** [comment]
I think the PR needs to be rebased or main merged into it to use fixed CI and have `workflow` check pass.

Code coverage check will probably fail because code coverage will decrease as there is quite a few lines not covered by unit tests (#80), but that could be ignored for now...

**Pull request merged**
